apps/entry/tests/test_entry_comment.py

```diff
@@ -0,0 +1,287 @@
+        data['assignee'] = self.user.pk
+
+        # Throw error if text is not provided
+        data['text'] = None
+        response = self.client.post(url, data)
+        self.assert_400(response)
+
+    def test_create_comment_reply(self):
+        entry_2 = self.create_entry()
+
+        url = '/api/v1/entry-comments/'
+        data = {
+            'entry': entry_2.pk,
+            'assignee': self.user.pk,
+            'text': 'This is first comment',
+            'parent': self.comment.pk,
+        }
+
+        self.authenticate()
+        response = self.client.post(url, data)
+        self.assert_201(response)
+        r_data = response.json()
+        assert r_data['entry'] == self.entry.pk, 'Should be same to parent entry'
+        assert r_data['assignee'] is None, 'There should be no assignee in reply comment'
+
+    def test_comment_text_history(self):
+        url = '/api/v1/entry-comments/'
+        data = {
+            'entry': self.entry.pk,
+            'assignee': self.user.pk,
+            'text': 'This is first comment',
+            'parent': None,
+        }
+
+        self.authenticate()
+        response = self.client.post(url, data)
+        self.assert_201(response)
+        comment_id = response.json()['id']
+
+        # Patch new text
+        new_text = 'this is second comment'
+        response = self.client.patch(f'{url}{comment_id}/', {'text': new_text})
+        r_data = response.json()
+        assert r_data['text'] == new_text
+        assert len(r_data['textHistory']) == 2
+
+        # Patch same text again
+        response = self.client.patch(f'{url}{comment_id}/', {'text': new_text})
+        r_data = response.json()
+        assert r_data['text'] == new_text
+        assert len(r_data['textHistory']) == 2
+
+    def test_comment_resolve(self):
+        url = '/api/v1/entry-comments/'
+        data = {
+            'entry': self.entry.pk,
+            'assignee': self.user.pk,
+            'text': 'This is first comment',
+            'parent': None,
+        }
+
+        self.authenticate(self.root_user)
+        # Add comment
+        response = self.client.post(url, data)
+        r_data = response.json()
+        self.assert_201(response)
+        assert r_data['isResolved'] is False
+        parent_comment_id_1 = r_data['id']
+
+        self.authenticate(self.user)
+        # Add comment
+        response = self.client.post(url, data)
+        r_data = response.json()
+        self.assert_201(response)
+        assert r_data['isResolved'] is False
+        parent_comment_id = r_data['id']
+
+        # Add reply comment
+        data['parent'] = parent_comment_id
+        response = self.client.post(url, data)
+        r_data = response.json()
+        self.assert_201(response)
+        assert r_data['isResolved'] is False
+        comment_id = r_data['id']
+
+        # Throw error if resolved request is send for reply
+        response = self.client.post(f'{url}{comment_id}/resolved/')
```

Comment:
Betterr to have `resolve` instead of `resolved` in url if it is an action.

Reply:
Done